### ISSUE TITLE
Add a concept of account

### ISSUE THREAD

**Issue by truedat101** [IoTone/TheShellProject#5]
An account needs:

(borrowed from NEM white paper)

- account balance
- number of harvested blocks
- height of the first transaction that referenced the account
- metadata: use this for any misc account info that we might determine we need later
- importance rank (for POI)
- vested balance (for POI)

Presume that each node can host up to N accounts.

**Comment by truedat101** [comment]
I'm assuming accounts can have addresses associated.  See #6 for more details.

**Comment by truedat101** [comment]
Reference: https://docs.nem.io/en/address-components#account-data

**Comment by truedat101** [comment]
Moving to proto3.

**Comment by pedroalvesbatista** [comment]
Right, as long Keypair is hashed to SHA-3, and can be verified, getting all to the address using the 6 steps from #9 , accounts is ready to go.

**Comment by truedat101** [comment]
en fuego ! 🔥

**Comment by truedat101** [comment]
@pedroalvesbatista please mark for "Needs Review" when it is ready for a review.

**Comment by pedroalvesbatista** [comment]
I added some new checks, but not so concrete, but working on it. After having general hash I can go on into account. But, the basics of balance and hervested blocks I putted the variables and methods (only the skeleton).

**Comment by pedroalvesbatista** [comment]
Seems that here we need to work properly on the struct to hold the account.

**Comment by truedat101** [comment]
The right approach is to have a default constructor for starts, one that zeroes at initial values.  Then any non-default versions that might be useful you can add as you go along.  Plenty of getter and setters will be useful.

**Comment by truedat101** [comment]
So account core is a wasteland right now.  Nothing done there.  I'll flesh this out.

**Comment by truedat101** [comment]
I'm going to mark the Struct as Depr (deprecated).

**Comment by truedat101** [comment]
Also - we are missing a wallet concept.  Add that for next iteration.

**Comment by truedat101** [comment]
Not sure what nembex vs network balance is: http://chain.nem.ninja/#/account/NCKZD7JGDLNDIVVPH6U2PG2QKD3PX3FX4CPZMF2A/0

Need to worry about that later.

**Comment by truedat101** [comment]
Please use shelld.accountcore.Account for the account holder info.
